examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -240,10 +251,14 @@ void OTAProviderExample::SendQueryImageResponse(app::CommandHandler * commandObj
         FabricInfo * fabricInfo = Server::GetInstance().GetFabricTable().FindFabricWithIndex(fabricIndex);
         NodeId nodeId           = fabricInfo->GetPeerId().GetNodeId();
 
-        // Only supporting BDX protocol for now
-        MutableCharSpan uri(uriBuf, kUriMaxLen);
-        chip::bdx::MakeURI(nodeId, CharSpan::fromCharString(mOTAFilePath), uri);
-        ChipLogDetail(SoftwareUpdate, "Generated URI: %.*s", static_cast<int>(uri.size()), uri.data());
+        // Generate the ImageURI if one is not already preset
+        if (strlen(mImageUri) == 0)
+        {
+            // Only supporting BDX protocol for now
+            MutableCharSpan uri(mImageUri, kUriMaxLen);
+            chip::bdx::MakeURI(nodeId, CharSpan::fromCharString(mOTAFilePath), uri);
```

Comment:
Pre-existing, but what guarantees null-termination here?  Code later relies on `mImageUri` being null-terminated...

Reply:
Let me make sure that `MakeURI` constructs a null-terminated value

Reply:
Updated the function header to indicate so